docs/05-engineering/systems-sandboxes.md

```diff
@@ -1,9 +1,290 @@
+
+#### /etc/hosts
+When installing a new site, (e.g., named "sitename") add a line to your /etc/hosts file like: 
+`127.0.0.1   sitename.localhost`
+Then when you visit `http://sitename.localhost` Apache looks for the (virtual) document root (using the enabled vhost_alias module) along the path defined in zz-localhost.conf and finds it in WORKSPACE/sitename/docroot/. See comments in that file (mentioned above and installed below in the "drush aliases section") for more information.
+
+#### Permissions
+Apache, by default, usually runs under the user:group permissions of www-data:www-data. In most cases, this is overly restrictive for personal development machines that aren't living a second life as a public server. The easiest way to get Apache to read the files in your ~/workspace/ directory is to change the permissions that Apache runs under to match your default user permissions. On Ubuntu, there is a file /etc/apache2/envvars that contains the variables `APACHE_RUN_USER` and `APACHE_RUN_GROUP`. Change the values these are set to from `www-data` to those that match your user account, and restart Apache.
+
+If you can't do this for some reason (e.g., your development box is also a public server) you can grant www-data group write permissions to the group that owns your ~/workspace/ files, but this ends up being a bit more fiddley.
+
+### MySQL
+
+#### /etc/mysql/my.cnf
+The Percona Configuration Wizard for MySQL does a good job.
+
+$HOME/.my.cnf
+Create a text file in your home directory named .my.cnf that contains the following (changing the user and password values to match your databas user and password names):
+
+```
+  [client]
+  host=localhost
+  user=root
+  password=root
+```
+
+Then when you run mysql from the command line, you'll no longer have to type in pesky usernames and passwords!
+
+#### Database naming
+We often have multiple sites running on our sandox machines, and multiple databases, so we've developed a standard naming convention that some tools expect.
+
+* sitename_drupal -- Drupal database for sitename (in your sandbox)
+* sitename_civicrm -- CiviCRM database for sitename (in your sandbox)
+* live_drupal -- Drupal database on LIVE (production) site (running on server vhost)
+* qa_drupal --Drupal database on QA site  (running on server vhost)
+* dev_drupal -- Drupal database on DEV (running on server vhost)
+
+#### MySQL Workbench
+Check out the [MySQL Workbench](https://www.mysql.com/products/workbench/) for a handy GUI - and there are others.
+(Which should be our primary recommendation?)
+
+### PHP
+/etc/php5/apache/php.ini
+There are many possible tweaks, but here are some you may want to consider:
+
+max_execution_time = 600
+memory_limit = 256M
+display_errors = On
+
+## Additional Tools
+
+### Composer
+a PHP dependency manager / package installer
+Install composer <https://getcomposer.org/doc/00-intro.md#installation-nix>
+
+### curl
+-sS <https://getcomposer.org/installer> | php
```

Comment:
Also may want to quote this to differentiate between commands and text in this file